packages/component/src/provider.ts

```diff
@@ -38,6 +38,7 @@ export class Provider<T extends Name> {
     string,
     Deferred<NameServiceMapping[T]>
   > = new Map();
+  private readonly instancesOptions: Map<string, InitializeOptions> = new Map();
```

Comment:
```suggestion
  private readonly instancesOptions: Map<string, Record<string, unknown>> = new Map();
```

`InitializeOptions` contains `instanceIdentifier` and `options`. Here we only wants the `options`.

Reply:
Changed.